gfx.cpp

```diff
@@ -5160,9 +5290,10 @@ class GfxInternal
     void collect(AccelerationStructure const &acceleration_structure)
     {
         destroyBuffer(acceleration_structure.bvh_buffer_);
-        for(size_t i = 0; i < acceleration_structure.raytracing_primitives_.size(); ++i)
-            if(raytracing_primitive_handles_.has_handle(acceleration_structure.raytracing_primitives_[i].handle))
-                destroyRaytracingPrimitive(acceleration_structure.raytracing_primitives_[i]);
+        auto copy = acceleration_structure.raytracing_primitives_;
```

Comment:
Now a copy is required since `destroyRaytracingPrimitive` call might remove elements